pkg/operator/controllers/node/node_controller.go

```diff
@@ -35,12 +35,13 @@ const (
 // NodeReconciler spots nodes that look like they're stuck upgrading.  When this
 // happens, it tries to drain them disabling eviction (so PDBs don't count).
 type NodeReconciler struct {
-	log           *logrus.Entry
+	log *logrus.Entry
+
 	arocli        aroclient.Interface
 	kubernetescli kubernetes.Interface
 }
 
-func NewNodeReconciler(log *logrus.Entry, kubernetescli kubernetes.Interface, arocli aroclient.Interface) *NodeReconciler {
+func NewReconciler(log *logrus.Entry, arocli aroclient.Interface, kubernetescli kubernetes.Interface) *NodeReconciler {
```

Comment:
Let's keep `NewNodeReconciler` as `NewReconciler` might mean that the return value is `reconcile.Reconciler` interace from controller-runtime, not `NodeReconciler`. Our reconciler meets the interface, but we return a specific value (whch is not idea, but a separate story). So I think to avoid confusion, for now we should follow `New(TypeWhichIsActuallyBeingReturned)` pattern.

Reply:
This suggestion contradicts [your other one](https://github.com/Azure/ARO-RP/pull/1602#discussion_r668895329). If we want to just use the type name `Reconciler`, inferring the type of reconciler from the package name, then we should also have the `New` function just be called `NewReconciler`. It will be clear enough that it is not `reconcile.Reconciler` due to the difference in package name.

Reply:
Yes, sorry. My comment was confusing indeed. It was based on content of this file at the time: the type at the moment of writing was called `NodeReconciler`, so I suggested to keep the function to be `NewNodeReconciler` assuming that the type name doesn't change.